### PR TITLE
[MI-3035]: Converted the get assignment API into GraphQL

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -966,19 +967,15 @@ func (p *Plugin) createIssueComment(c *UserContext, w http.ResponseWriter, r *ht
...
-	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
+	graphQLClient := p.graphQLConnect(c.GHInfo)
 
-	username := c.GHInfo.GitHubUsername
-	query := getYourAssigneeSearchQuery(username, config.GitHubOrg)
-	result, _, err := githubClient.Search.Issues(c.Ctx, query, &github.SearchOptions{})
+	assignmentResponse, err := graphQLClient.Issues.GetYourAssignment()
 	if err != nil {
-		c.Log.WithError(err).With(logger.LogContext{"query": query}).Warnf("Failed to search for assignments")
+		c.Log.WithError(err).Warnf("Failed to search for PRs")
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go]: Will change this to assignment in the next commit

File: server/plugin/internal/graphql/assignment.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		queryParamAssignmentCursor:      (*githubv4.String)(nil),
+	}
+
+	if i.client.org != "" {
+		params[queryParamAssignmentSearchQuery] = githubv4.String(fmt.Sprintf("org:%s %s", i.client.org, params[queryParamAssignmentSearchQuery]))
+	}
+
+	var query assignmentSearchQuery
+	var res []*github.Issue
+
```
Comment (raghavaggarwal2308) [inline server/plugin/internal/graphql/assignment.go]:
```suggestion
```

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by raghavaggarwal2308** [comment]
Check failing CI

File: server/plugin/internal/graphql/assignment.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+					},
+					HTMLURL: &htmlURL,
+				}
+			} else {
+				prNumber := int(resp.PullRequest.Number)
+				repositoryURL := resp.PullRequest.Repository.URL.String()
+				htmlURL := resp.PullRequest.URL.String()
+				title := string(resp.PullRequest.Title)
+				createdAt := resp.PullRequest.CreatedAt.Time
+				updatedAt := resp.PullRequest.UpdatedAt.Time
```
Comment (ayusht2810) [inline server/plugin/internal/graphql/assignment.go]: we can reduce the line of codes here

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/api.go [hunk 4]
```diff
@@ -966,19 +967,15 @@ func (p *Plugin) createIssueComment(c *UserContext, w http.ResponseWriter, r *ht
...
-	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
+	graphQLClient := p.graphQLConnect(c.GHInfo)
 
-	username := c.GHInfo.GitHubUsername
-	query := getYourAssigneeSearchQuery(username, config.GitHubOrg)
-	result, _, err := githubClient.Search.Issues(c.Ctx, query, &github.SearchOptions{})
+	assignmentResponse, err := graphQLClient.Issues.GetYourAssignment()
 	if err != nil {
-		c.Log.WithError(err).With(logger.LogContext{"query": query}).Warnf("Failed to search for assignments")
+		c.Log.WithError(err).Warnf("Failed to search for Assignments")
```
Comment (manojmalik20) [inline server/plugin/api.go:974]:
```suggestion
		c.Log.WithError(err).Warnf("Failed to search for assignments")
```

File: server/plugin/internal/graphql/assignment.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+const (
+	queryParamAssignmentCursor      = "assignmentCursor"
+	queryParamAssignmentSearchQuery = "assignmentSearchQueryArg"
+)
+
+// IssueService is responsible for handling issue related queries
+type IssueService service
+
+func (i *IssueService) GetYourAssignment() ([]*github.Issue, error) {
```
Comment (manojmalik20) [inline server/plugin/internal/graphql/assignment.go:19]:
```suggestion
func (i *IssueService) GetYourAssignments() ([]*github.Issue, error) {
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request closed without merging**